erc20/src/lib.rs

```diff
@@ -36,6 +39,12 @@ impl ERC20 {
         if owner != address!("0000000000000000000000000000000000000007") {
             revert();
         }
+        if msg_value() != U256::from(0) {
+            revert();
+        }
+        if msg_sig() != [2, 0, 0, 0] {
+            revert();
+        }
```

Comment:
Should we enforce deterministic 4-byte discriminants here from the function signature, with the ability to override via a `#[sig(0x12345678)` attribute?

Reply:
yep makes sense

Reply:
Yes, have an issue for it here: https://github.com/r55-eth/r55/issues/7

Reply:
and this too?